util/logutil/log.go

```diff
@@ -345,3 +347,30 @@ func SetLevel(level string) error {
 	zaplog.SetLevel(l.Level())
 	return nil
 }
+
+type correlationIDType int
+
+const (
+	ctxLogKey correlationIDType = iota
+)
+
+// Logger gets a contextual logger from current context.
+// contextual logger will output common fields from context.
+func Logger(ctx context.Context) *zap.Logger {
+	if ctxlogger, ok := ctx.Value(ctxLogKey).(*zap.Logger); ok {
+		return ctxlogger
+	}
+	return plog.L()
+}
+
+// WithConnID attaches connId to context.
+func WithConnID(ctx context.Context, connID uint32) context.Context {
+	return context.WithValue(ctx, ctxLogKey, plog.L().With(zap.Uint32("connID", connID)))
+}
+
+// WithRecvTs attaches current packet received timestamp to context.
+// except load data, it's common that sql to one request packet in mysql protocol,
+// so we can use recvTs to gather log for some sql request on one connection.
+func WithRecvTs(ctx context.Context, recvTs int64) context.Context {
+	return context.WithValue(ctx, ctxLogKey, plog.L().With(zap.Int64("recvTs", recvTs)))
```

Comment:
Do we need to change the key（`ctxLogKey `）？

Reply:
It should be `return context.WithValue(ctx, ctxLogKey, Logger(ctx).With(zap.Int64("recvTs", recvTs))`?